typescript/infra/scripts/funding/vanguard.ts

```diff
@@ -0,0 +1,233 @@
+
+  // Track which vanguards were topped up
+  const topUpsNeeded: ChainMap<VanguardBalance[]> = {};
+
+  await Promise.all(
+    VANGUARD_NETWORKS.map(async (chain) => {
+      for (const vanguard of VANGUARDS) {
+        const address = VANGUARD_ADDRESSES[vanguard];
+        try {
+          const provider = multiProvider.getProvider(chain);
+          const currentBalance = await provider.getBalance(address);
+          const desiredBalance = parseUnits(
+            VANGUARD_FUNDING_CONFIGS[chain][vanguard],
+            TOKEN_DECIMALS,
+          );
+          const delta = desiredBalance.sub(currentBalance);
+
+          if (delta.gt(MIN_FUNDING_AMOUNT)) {
+            topUpsNeeded[chain] = topUpsNeeded[chain] || [];
+            topUpsNeeded[chain].push({
+              chain,
+              vanguard,
+              balance: Number(formatUnits(delta, TOKEN_DECIMALS)).toFixed(3),
+            });
+          }
+        } catch (error) {
+          rootLogger.error(
+            chalk.bold.red(
+              `Error topping up ${vanguard} on chain ${chain}:`,
+              error,
+            ),
+          );
+        }
+      }
+    }),
+  );
+
+  // Print summary of topped up vanguards
+  if (Object.keys(topUpsNeeded).length > 0) {
+    rootLogger.info('\nTop ups needed for the following:');
+    // eslint-disable-next-line no-console
+    console.table(
+      Object.entries(topUpsNeeded).reduce((acc, [chain, topUps]) => {
+        const chainEntries: Record<VanguardName, string> = {} as Record<
+          VanguardName,
+          string
+        >;
+        VANGUARDS.forEach((vanguard) => {
+          const match = topUps.find((t) => t.vanguard === vanguard);
+          chainEntries[vanguard] = match ? match.balance : '-';
+        });
+        acc[chain] = chainEntries;
+        return acc;
+      }, {} as Record<string, Record<VanguardName, string>>),
+    );
+
+    if (fund) {
+      rootLogger.info(chalk.italic.blue('\nFunding vanguards...'));
+    } else {
+      rootLogger.info(chalk.italic.yellow('\nDry run - not funding vanguards'));
+      return;
+    }
+
+    await Promise.all(
+      Object.entries(topUpsNeeded).map(([chain, topUps]) =>
+        topUps.map(async ({ vanguard, balance }) => {
+          try {
+            const signer = multiProvider.getSigner(chain);
+            const signerBalance = await signer.getBalance();
+            const amount = BigNumber.from(balance);
+
+            if (signerBalance.lt(amount)) {
+              rootLogger.warn(
+                chalk.bold.yellow(
+                  `Insufficient balance for ${vanguard} on ${chain}. Required: ${formatUnits(
+                    amount,
+                    TOKEN_DECIMALS,
+                  )}, Available: ${formatUnits(signerBalance, TOKEN_DECIMALS)}`,
+                ),
+              );
+              return;
+            }
+
+            return multiProvider.sendTransaction(chain, {
+              to: VANGUARD_ADDRESSES[`${vanguard}`],
```

Comment:
why not just use vanguard directly?

Reply:
again was from an old version before i fixed the types